components/image-builder-mk3/pkg/orchestrator/orchestrator.go

```diff
@@ -334,8 +336,15 @@ func (o *Orchestrator) Build(req *protocol.BuildRequest, resp protocol.ImageBuil
 		bobBaseref += ":latest"
 	}
 	wsref, err := reference.ParseNamed(wsrefstr)
+	var workspaceAuth []byte
 	var additionalAuth []byte
 	if err == nil {
+		res := make(auth.ImageBuildAuth)
+		res[reference.Domain(wsref)] = registry.AuthConfig(*wsrefAuth)
```

Comment:
`wsrefAuth` can very much be nil here which would nil-deref. Please add a nil check.